File: csharp/twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> hashMap = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            int currentNum = nums[i];
            int difference = target - currentNum;
            if(hashMap.ContainsKey(difference))
            {
                return new int[] { hashMap[difference], i };
            }
            hashMap[currentNum] = i;
        }
        return Array.Empty<int>();
    }
}

File: js/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let result;
  const hashMap = {};
  for(let i = 0; i <= nums.length; i++){
      const num = nums[i]
      const difference = target - num;
      console.log(JSON.stringify(hashMap), difference, num)
      if(difference in hashMap)
          result = [i, hashMap[difference]]
      hashMap[num] = i;
  }
  return result;
};
